### DISCUSSION TITLE
Utilisation d'un régulateur LSA-H3P50YB avec l'entrée PWM

### DISCUSSION THREAD

**Discussion by PSA38** [mathieucarbou/YaSolR#32] [General & Support]
Bonjour,
Est-il possible d'utiliser l'entrée PWM du régulateur de tension LSA-H3P50YB (fréquence entre 1 et 3kHz, niveau 4 à 24Vdc) sans avoir besoin du ZC puisque la fonction ZC est déjà intégrée dans régulateur ?

Merci pour votre aide.

Cordialement

**Comment by mathieucarbou** [comment]
Oui, le code est là mais il faut que je teste encore, mais en théorie ça devrait marcher. Par l'entrée 0-10V par contre.

Sur mon blog (https://yasolr.carbou.me/blogs), dans les examples de routeur Shelly et Home Assistant, je contrôle le LSA via le Shelly dimmer gen 3, qui permet de dimmer entre 0-10V. Le LSA a une entrée analogique 0-10, mais elle n'est pas très linéaire (pas très grave pour le routage car l'algonquine PID devrait s'ajuster).

Donc l'idée est d'utiliser un convertisseur PWM analogique (j'en liste sur mon site), qui vont permettre de convertir un duty cycle généré par l'ESP32, en une tension entre 0-10V.

Comme le LSA fonctionne en réalité entre environ 1-8V, il faut, tout comme ave le Shelly dimmer, faire ce qu'on appelle un remapping de la plage 0-100%. Afin que le nouveau 0% soit 10% et le 100% deviennent 85% par exemple. 

Tout ça est expliqué dans le manuel :-)

Le routeur offgrid PVBrain utilise ce même mécanisme aussi.

Je n'ai juste pas eu le temps encore de tester sur YaSolR, c'est dans ma todo.

Cette solution requiert une alim 12V par contre pour alimenter le module de conversion + LSA en plus (i.e. un Meanwell HDR-15 apr exemple)

**Comment by PSA38** [comment]
Merci pour votre réactivité. L’utilisation de l’entrée PWM permet de simplifier le routeur, il faut simplement ajouter un transistor connecté au +5V et commander sa base par la sortie GPIO PWM de l’ESP32. Pas besoin d’alimentation 12V, ni de convertisseur DAC, donc plus simple et plus économique. Votre solution logiciel de routeur est super ! Je vais la tester dés réception de mon LSA. Cordialement.  De : Mathieu CarbouEnvoyé le :lundi 7 octobre 2024 17:21À : mathieucarbou/YaSolRCc : PSA38; AuthorObjet :Re: [mathieucarbou/YaSolR] Utilisation d'un régulateur LSA-H3P50YB avec l'entrée PWM (Discussion #32) Oui, le code est là mais il faut que je teste encore, mais en théorie ça devrait marcher. Par l'entrée 0-10V par contre.Sur mon blog (https://yasolr.carbou.me/blogs), dans les examples de routeur Shelly et Home Assistant, je contrôle le LSA via le Shelly dimmer gen 3, qui permet de dimmer entre 0-10V. Le LSA a une entrée analogique 0-10, mais elle n'est pas très linéaire (pas très grave pour le routage car l'algonquine PID devrait s'ajuster).Donc l'idée est d'utiliser un convertisseur PWM analogique (j'en liste sur mon site), qui vont permettre de convertir un duty cycle généré par l'ESP32, en une tension entre 0-10V.Comme le LSA fonctionne en réalité entre environ 1-8V, il faut, tout comme ave le Shelly dimmer, faire ce qu'on appelle un remapping de la plage 0-100%. Afin que le nouveau 0% soit 10% et le 100% deviennent 85% par exemple. Tout ça est expliqué dans le manuel.Le routeur offgrid PVBrain utilise ce même mécanisme aussi.Je n'ai juste pas eu le temps encore de tester sur YaSolR, c'est dans ma todo.Cette solution requiert une alim 12V par contre pour alimenter le module de conversion + LSA en plus (i.e. un Meanwell HDR-15 apr exemple)—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by mathieucarbou** [comment]
Effectivement si vous avez une solution pour faire de la conversion pwm c'est l'idéal mais je ne vois vraiment pas comment.

L'esp32 sort un signal pwm en 3.3V. donc même si vous faites une conversion pwm d'une plage de fréquence à une autre, il n'en reste pas moins que le LSA requiert un signal pwm sur 5V alors que l'esp sort du 3.3V..
 
Le pwm en sortie de l'esp est aussi sur 50 Hz ou 60 Hz alors que le LSA a une fréquence d'entrée entre 1-3kHz...

J'avais regardé avec un ami, les gars de pvbrain aussi on essayé, et la conclusion a été qu'il est plus facile de passer par l'analogique 0-10V.

Tenez moi au courant car si vous avez un setup en place sur lequel on peut tester ça, je suis tout à fait ouvert à le supporter si ça marche! 

Le seul problème c'est que ça requiert la réécriture d'une librairie de dimmer spécialisée pour ça, alors que passe par un module de conversion pwm à analogique permet de réutiliser le même code, mais on met in LSA à la place d'un SSR ou triac.

**Comment by mathieucarbou** [comment]
Je viens de tester ce soir le support du LSA, et ça fonctionne super bien! Même pas besoin d'utiliser la fonctionnalité de remapping lorsqu'on utilise le module de conversion PWM à 0-10V (analogique) qui est listé sur le site.

J'ai mis à jour la doc avec quelques captures oscilloscope:

https://yasolr.carbou.me/manual#voltage-regulators-lsa-lctc-etc
